### ISSUE TITLE
macOS is blocking macFuse even after it is approved

### ISSUE THREAD

**Issue by Chasapple4** [osxfuse/osxfuse#779]
macOS 11.1 is blocking macFUSE 4.0.5 but not macFUSE 4.0.2 (have to downgrade, can confirm as VirtualBox Win 10 guests will crash and XP guests are semi stable in Virtualbox 6.1.18 with macFUSE 4.0.5 but are fine with 4.0.2)

**Comment by bfleischer** [comment]
> macOS 11.1 is blocking macFUSE 4.0.5

macFUSE does not interfere with VirtualBox in any way. If macOS is not loading the 4.0.5 kernel extension, how can macFUSE 4.0.5 cause issues with VirtualBox?

> macOS is blocking macFuse even after it is approved

Are you prompted to approve the kernel extension again? If this is the case, you are running into a bug in macOS 11. Apple re-wrote the Kext Manager component, that is responsible for loading kernel extensions in macOS 11. As with all new software there a bugs. Apple classified this issue as critical and is working on a fix.

In the meantime, try the following workaround:  Install macFUSE 4.0.5. Then run the following command in Terminal after booting into the recovery environment:

kmutil trigger-panic-medic --volume-root /Volumes/System

This disables loading third-party kernel extensions until you approve them again. Then reboot your Mac and try to use macFUSE 4.0.5. macOS will prompt you again to approve loading the kernel extension. Follow the instruction and then reboot your Mac again.

**Comment by Chasapple4** [comment]
VirtualBox uses macFuse for many of the guest OSes now, macFuse 4.0.5 worked with 6.1.16 just fine, after 6.1.18 it did not.  A previous versions VirtualBox I had to revert to macFuse 4.0.2 for any Windows guest VM to load.
`https://www.virtualbox.org/manual/UserManual.html#vboximg-mount
https://www.virtualbox.org/manual/UserManual.html#licOsxFuse  
Oracle does not fully list in the user manual what all macFuse is used for`

VirtualBox 6.1.18 has an older version of macFuse in the installer (a non macOS 11 compatible version OSXFuse 3.10.4) so I do custom to not install the older version from Dec 2019`
Win 10 20H2 guest crashes with macFuse blocked, while dev insider branch builds would just be in a blue screen loop (installed macFuse 4.0.2 over 4.0.5 and that allowed both to boot normally, XP kinda booted with macFuse 4.0.5 blocked by macOS but some apps were unstable)

I had already approved the extension and restarted and it had been loaded fine.  Then randomly macOS blocked macFuse from loading with no visible notice.  There is no notice in the Security preference pane to approve it again (it is a silent blocking of the system extension), install  macFuse 4.0.2 over 4.0.5 and it loads fine (don't even need to restart for macFuse to load again).

Do you have a public bug number (or note where Apple has confirmed the bug) for the system extension loading bug?  (strange is that only 2 systems extensions are affected by the bug while others are not, macFuse is one of them)

**Comment by Chasapple4** [comment]
macOS 11.2 looks to be blocking both 4.0.2 & 4.0.5 of macFuse
kmutil --trigger-panic-medic --volume-root /Volumes/System did not work

Is there a terminal command you know of to force macOS to load the System Extension?  (I have 2 that are already approved but macOS is blocking months after they were approved)

**Comment by bfleischer** [comment]
macOS is not blocking macFUSE per se. As I explained before Apple's new Kext Manager implementation on macOS 11 has some bugs and enters an inconsistent state from time to time.  All you can do is uninstall macFUSE, triggering the panic medic, reboot your Mac, reinstall macFUSE and try again. Reinstalling macOS might work, too. There is very little that can be done from the macFUSE end, Apple needs to address the Kext Manager bugs. 

macFUSE may be used for mounting virtual disks, but it is not responsible for Windows not booting up in your virtual machine. macFUSE is not loaded in the context of your virtual machine. The crashes and blue screens you are observing might be related to loading VirtualBox kernel extensions on macOS or could point to completely unrelated issues. In any case, please raise theses issues with the VirtualBox team.

**Comment by Chasapple4** [comment]
Strange that worked.  Do you have any info on the bug report? (I think it affects both macOS 10.15.x and macOS 11.2), don't want to have to redo the uninstall and  recovery mode fix every days/ weeks.

Should this be changed to waiting on a fix from Apple for the label?

**Comment by bfleischer** [comment]
macOS 10.15 should be fine. The kernel extension loading issues are a result of bugs that were introduced in macOS 11. Apple is investigating the issues. I do not know when Apple will release the fix.

**Comment by Chasapple4** [comment]
I was seeing the bug on macOS 10.15.7 also (tho not as bad) with a different system extension that was already approved, but was blocked in early December 2020.  Then I saw it a bit in 11.0.1 and in 11.1 with MacFuse not loading at all and sometimes would only load with the 4.0.2 prerelease, and higher versions would not load.

**Comment by Chasapple4** [comment]
After a few restarts of the Mac looks to be blocked again

**Comment by Chasapple4** [comment]
Found this in the console (when I tried to boot a Win 10 guest VM)
`2021-02-09 14:08:33.145059 -0700	default	14:08:33.145059-0700	kernelmanagerd	validating extension at /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext
2021-02-09 14:08:33.146804 -0700	info	14:08:33.146804-0700	kernelmanagerd	Collision: replacing Kext io.macfuse.filesystems.macfuse v2053.20 in executable kext bundle io.macfuse.filesystems.macfuse at /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext
2021-02-09 14:08:34.360673 -0700	default	14:08:34.360673-0700	kernelmanagerd	checking bundle path /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext for Kext io.macfuse.filesystems.macfuse v2053.20 in executable kext bundle io.macfuse.filesystems.macfuse at /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext
2021-02-09 14:08:34.655024 -0700	default	14:08:34.655024-0700	kernelmanagerd	validating extension at /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext
2021-02-09 14:08:34.656632 -0700	info	14:08:34.656632-0700	kernelmanagerd	Collision: replacing Kext io.macfuse.filesystems.macfuse v2053.20 in executable kext bundle io.macfuse.filesystems.macfuse at /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext
2021-02-09 14:08:34.728418 -0700	default	14:08:34.728418-0700	kernelmanagerd	checking bundle path /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext for Kext io.macfuse.filesystems.macfuse v2053.20 in executable kext bundle io.macfuse.filesystems.macfuse at /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext
`

**Comment by Chasapple4** [comment]
Ran the installer again (on macOS 11.2.1)
`2021-02-09 14:11:19.541294 -0700	default	14:11:19.541294-0700	kernel	hfs: mounted macFUSE on device disk2
2021-02-09 14:11:19.702014 -0700	info	14:11:19.702014-0700	sharedfilelistd	[<private>] resolving item: macFUSE [D81832C2-B7E1-4055-A5DF-68BE47F136FD] - URL:/Volumes/macFUSE options: 768
2021-02-09 14:11:19.704097 -0700	info	14:11:19.704097-0700	sharedfilelistd	[<private>] resolving item: macFUSE [D81832C2-B7E1-4055-A5DF-68BE47F136FD] - URL:/Volumes/macFUSE options: 768
2021-02-09 14:11:50.195364 -0700	error	14:11:50.195364-0700	Finder	validatedCachedResults: unable to create keyName from: {
    "CACHE_DELETE_URGENCY" = 3;
    "CACHE_DELETE_VOLUME" = "/Volumes/macFUSE";
}
2021-02-09 14:11:50.197495 -0700	error	14:11:50.197495-0700	Finder	validatedCachedResults: unable to create keyName from: {
    "CACHE_DELETE_URGENCY" = 3;
    "CACHE_DELETE_VOLUME" = "/Volumes/macFUSE";
}
2021-02-09 14:11:50.240092 -0700	error	14:11:50.240092-0700	Finder	validatedCachedResults: unable to create keyName from: {
    "CACHE_DELETE_URGENCY" = 3;
    "CACHE_DELETE_VOLUME" = "/Volumes/macFUSE";
}
2021-02-09 14:11:50.271683 -0700	error	14:11:50.271683-0700	Finder	validatedCachedResults: unable to create keyName from: {
    "CACHE_DELETE_URGENCY" = 3;
    "CACHE_DELETE_VOLUME" = "/Volumes/macFUSE";
}
2021-02-09 14:11:50.303138 -0700	error	14:11:50.303138-0700	Finder	validatedCachedResults: unable to create keyName from: {
    "CACHE_DELETE_URGENCY" = 3;
    "CACHE_DELETE_VOLUME" = "/Volumes/macFUSE";
}
2021-02-09 14:12:00.210390 -0700	default	14:12:00.210390-0700	lsd	Non-fatal error enumerating at <private>, continuing: Error Domain=NSCocoaErrorDomain Code=260 "The file “PlugIns” couldn’t be opened because there is no such file." UserInfo={NSURL=PlugIns/ -- file:///Library/Filesystems/macfuse.fs/Contents/Resources/uninstall_macfuse.app/Contents/, NSFilePath=/Library/Filesystems/macfuse.fs/Contents/Resources/uninstall_macfuse.app/Contents/PlugIns, NSUnderlyingError=0x7fa6e6fb14f0 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
2021-02-09 14:12:00.222461 -0700	default	14:12:00.222461-0700	lsd	Non-fatal error enumerating at <private>, continuing: Error Domain=NSCocoaErrorDomain Code=260 "The file “PlugIns” couldn’t be opened because there is no such file." UserInfo={NSURL=PlugIns/ -- file:///Library/Filesystems/macfuse.fs/Contents/Resources/uninstall_macfuse.app/Contents/, NSFilePath=/Library/Filesystems/macfuse.fs/Contents/Resources/uninstall_macfuse.app/Contents/PlugIns, NSUnderlyingError=0x7fa6e6d40820 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
2021-02-09 14:12:00.682122 -0700	info	14:12:00.682122-0700	amfid	Entering OSX path for /Library/Filesystems/macfuse.fs/Contents/Resources/load_macfuse
2021-02-09 14:12:00.774923 -0700	default	14:12:00.774923-0700	kernelmanagerd	loadExtensions(withPaths:withIdentifiers:withPersonalityNames:options:withReply:): Loading paths: ["/Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext"], identifiers: [], personalityNames: []
2021-02-09 14:12:00.775995 -0700	info	14:12:00.775995-0700	kernelmanagerd		Kext io.macfuse.filesystems.macfuse v2053.20 in executable kext bundle io.macfuse.filesystems.macfuse at /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext
2021-02-09 14:12:01.028681 -0700	default	14:12:01.028681-0700	kernelmanagerd	validating extension at /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext
2021-02-09 14:12:01.030337 -0700	info	14:12:01.030337-0700	kernelmanagerd	Collision: replacing Kext io.macfuse.filesystems.macfuse v2053.20 in executable kext bundle io.macfuse.filesystems.macfuse at /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext
2021-02-09 14:12:01.385750 -0700	default	14:12:01.385750-0700	kernelmanagerd	checking bundle path /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext for Kext io.macfuse.filesystems.macfuse v2053.20 in executable kext bundle io.macfuse.filesystems.macfuse at /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext
2021-02-09 14:12:01.409759 -0700	default	14:12:01.409759-0700	kernelmanagerd	other kext requests: 
	Kext io.macfuse.filesystems.macfuse v2053.20 in executable kext bundle io.macfuse.filesystems.macfuse at /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext
2021-02-09 14:12:01.472686 -0700	default	14:12:01.472686-0700	kernelmanagerd	validating extension at /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext
2021-02-09 14:12:01.476865 -0700	info	14:12:01.476865-0700	kernelmanagerd	Collision: replacing Kext io.macfuse.filesystems.macfuse v2053.20 in executable kext bundle io.macfuse.filesystems.macfuse at /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext

	Kext io.macfuse.filesystems.macfuse v2053.20 in executable kext bundle io.macfuse.filesystems.macfuse at /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext 

		}
		TCCD_MSG_ALLOW_STANDARD_USER_TO_SET_SYSTEM_SERVICE=false
		auth_version=1 (0x1)
		auth_reason=0 (0x0)
		granted=false
		code_requirement=<data: 0x7fc256318b10>: { length = 148 bytes, contents = 0xfade0c000000009400000001000000060000000200000006... }
	}
	[1] = {
		auth_value=0 (0x0)
		path="/Library/Filesystems/macfuse.fs/<…>
2021-02-09 14:12:01.742361 -0700	default	14:12:01.742361-0700	syspolicyd	canLoadKernelExtension - direct, evaluate, Kext (3T5GSNBU6W, io.macfuse.filesystems.macfuse) version: ba597ad4dae8d4cb5bd2408c7b60dfe8e530a54e
2021-02-09 14:12:01.745595 -0700	default	14:12:01.745595-0700	syspolicyd	Kernel Extension ALLOWED: Kext (3T5GSNBU6W, io.macfuse.filesystems.macfuse) version: ba597ad4dae8d4cb5bd2408c7b60dfe8e530a54e
2021-02-09 14:12:01.772667 -0700	default	14:12:01.772667-0700	kernelmanagerd	approval results: 
	/Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext
2021-02-09 14:12:01.774192 -0700	default	14:12:01.774192-0700	kernelmanagerd	existing, approved auxiliary kext collection direct loads: 
	Kext io.macfuse.filesystems.macfuse v2053.20 in executable kext bundle io.macfuse.filesystems.macfuse at /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext
2021-02-09 14:12:01.774451 -0700	default	14:12:01.774451-0700	kernelmanagerd	existing, loadable, and approved auxiliary kext collection direct loads: 
	Kext io.macfuse.filesystems.macfuse v2053.20 in executable kext bundle io.macfuse.filesystems.macfuse at /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext
2021-02-09 14:12:01.777040 -0700	default	14:12:01.777040-0700	kernelmanagerd	new auxiliary kext collection candidates: 
	Kext io.macfuse.filesystems.macfuse v2053.20 in executable kext bundle io.macfuse.filesystems.macfuse at /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext (existing)
2021-02-09 14:12:01.777839 -0700	default	14:12:01.777839-0700	kernelmanagerd	direct loads: 
	Kext io.macfuse.filesystems.macfuse v2053.20 in executable kext bundle io.macfuse.filesystems.macfuse at /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext
2021-02-09 14:12:01.778084 -0700	default	14:12:01.778084-0700	kernelmanagerd	auxKC filesystem check: Kext io.macfuse.filesystems.macfuse v2053.20 in executable kext bundle io.macfuse.filesystems.macfuse at /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext found at /Library/Filesystems/macfuse.fs/Contents/Extensions/11/macfuse.kext
2021-02-09 14:12:01.778167 -0700	default	14:12:01.778167-0700	kernelmanagerd	loading extension from in-kernel fileset: io.macfuse.filesystems.macfuse (will start,  will match
2021-02-09 14:12:01.782284 -0700	default	14:12:01.782284-0700	kernelmanagerd	Received kext load notification: io.macfuse.filesystems.macfuse
2021-02-09 14:12:01.785684 -0700	info	14:12:01.785684-0700	opendirectoryd	UID: 501, EUID: 0, GID: 20, EGID: 20, PID: 1217, PROC: load_macfuse RPC: getgrnam, Module: SystemCache, rpc_version: 2, name: <private>
2021-02-09 14:12:01.786484 -0700	info	14:12:01.786484-0700	opendirectoryd	PID: 1217, Client: 'load_macfuse', exited with 0 session(s), 0 node(s) and 0 active request(s)
`

**Comment by Chasapple4** [comment]
Strange workaround that may or mayn't work, is that I ran the macFUSE 4.0.5 installer again and right after clicking close (when the install finished) opened the VMs I had been not having any luck with.

**Comment by Chasapple4** [comment]
Another reinstall, and wait about 12 hours and macOS decided to load it.

**Comment by ijoseph** [comment]
Here's my story: 
1. I had 11.1 (I believe) and had installed and was using 4.0.5 successfully. 
2. I updated to 11.2. At this point it claimed the extension was blocked; it prompted me to rebuild my kext cache and restart; after restarting, the same thing happened (apparently something in the rebuild/ restart process had failed). 
3. I updated to 11.3 Beta (20E5186d). Same issue. 
4. I manually [triggered a kernel panic in recovery mode](https://developer.apple.com/forums/thread/661240?login=true): `kmutil trigger-panic-medic --volume-root /Volumes/<YourVolumeName>`
5. After restarting, macFUSE 4.0.5 was loaded and worked successfully:
```bash
❯ kextstat -l | grep -v apple
No variant specified, falling back to release
  231    0 0xfffffe0006f18000 0x4000     0x4000     io.macfuse.filesystems.macfuse (2053.20) 9CF16245-CA95-3610-A0A2-AD85C8CD11C1 <7 5 4 3 1>
```

**Comment by bfleischer** [comment]
From what I have heard, the kernel extension loading issues should be fixed in an upcoming macOS 11.3 beta release.

**Comment by Chasapple4** [comment]
After macOS 11.2.3 update it was blocked again, ran the installer again 4.0.5 and then it loaded fine (no restart needed)

**Comment by fabrizziosoares** [comment]
I am new in this discussion, but I wonder if to install MacFuse and approve it, the only way is reducing security in startup tools. Is it possible to install MacFuse and keep high security?

**Comment by bfleischer** [comment]
@fabrizziosoares No, on M1 Macs you need to run macOS in "reduced security mode", otherwise third-party kernel extensions cannot be loaded. Your system is not any less secure than macOS 10.15 ever was.

@Chasapple4 The latest macOS release (macOS 11.3 beta 4) addresses the approve-restart loop bug.

**Comment by Chasapple4** [comment]
@bfleischer After the macOS 11.2.3 it worked as I mentioned after I ran the installer again, and then had to wait a few hours (tho I tried 12 hours later and things were working again).

**Comment by fabrizziosoares** [comment]
Thanks @bfleischer . Apple being Apple!

**Comment by slawek-es** [comment]
As a perhaps interesting data point, I found that macOS 11.2 will enter the "approve loop" for extensions which have been loaded from a directory under `/Applications`. After approving the kext and restarting, I was getting logs like:
```
2021-03-21 18:53:52.292879+0000 0x7b0      Default     0x0                  50     0    kcgend: Discarding io.macfuse.filesystems.macfuse: Validation failed: Instructions for io.macfuse.filesystems.macfuse in team 3T5GSNBU6W have different hash:  c63c13e0126c6f9731d6f5c72eebbe0d7b7cc42e b6e3d49f310245e8ec4502f55c67326dfe2e2b29
```
and
```
2021-03-21 19:54:10.397770+0100 0x69a      Default     0x0                  89     0    kernelmanagerd: kext io.macfuse.filesystems.macfuse was discarded: Validation failed: Instructions for io.macfuse.filesystems.macfuse in team 3T5GSNBU6W have different hash:  c63c13e0126c6f9731d6f5c72eebbe0d7b7cc42e b6e3d49f310245e8ec4502f55c67326dfe2e2b29
```
Obviously, this is a problem if you bundle your kernel extension along with your application.

As a workaround, I can copy the extension directory to some other location _before first loading_ and do a manual `kextload` from there. Literally any location, like directories under `/Users` and `/tmp`, worked for me. After approving the kext and rebooting, the `macfuse` kernel extension worked when triggered by `Mounter` from its location within our application bundle in `/Applications`.

I am going to try this out under macOS 11.3 Beta 4 and see if the upgrade fixes this problem as well.

**Comment by Chasapple4** [comment]
Update to [4.1.0 ](https://github.com/osxfuse/osxfuse/releases/tag/macfuse-4.1.0)and it worked right away

**Comment by bfleischer** [comment]
@slawek-es macOS 11.3 beta 4 fixes the issue.

**Comment by Chasapple4** [comment]
Rebooted last week and broken again.

How do you force macOS to load macFUSE?

**Comment by bfleischer** [comment]
@Chasapple4 Upgrade to macOS 11.3 beta 4 or later and the issue will disappear. There really is no other way.

**Comment by Chasapple4** [comment]
> @Chasapple4 Upgrade to macOS 11.3 beta 4 or later and the issue will disappear. There really is no other way.

Might be as last time after hours of waiting macOS did something and it was loaded.

**Comment by Chasapple4** [comment]
macOS 11.3 is out (around 6GB update), looks to have fixed it tho had to run the installer for VirtualBox again as the system extension was not loaded right by macOS

**Comment by Chasapple4** [comment]
@bfleischer Not fully fixed on 11.3, just got it where macOS is not loading macFuse.

**Comment by Chasapple4** [comment]
@bfleischer macOS 11.3.1 and it is still not loading.

Is there a way to do a full uninstall of macFuse and do a reinstall?

**Comment by slawek-es** [comment]
I can confirm what @Chasapple4 reported - macOS 11.3.1 does _not_ work with macFUSE 4.0.2 or 4.0.5, with the same issue as in 11.2. Meanwhile, version 11.4 Beta works fine. Looks like the fix was taken out at the last minute?

I am aware that this is none of your fault @bfleischer of course; but do you know any details behind the apparent regression?

**Comment by bfleischer** [comment]
macFUSE 4.1.0 (and EFS FUSE 4.1.0) work just fine for me on macOS 11.3.1 on Intel and Apple Silicon hardware.

**Comment by Chasapple4** [comment]
> macFUSE 4.1.0 (and EFS FUSE 4.1.0) work just fine for me on macOS 11.3.1 on Intel and Apple Silicon hardware.

Does it still work after you restart your computer?  (a restart and macOS will refuse to load it even after it has been approved, reinstall and the panic trigger from recovery do not work)

**Comment by bfleischer** [comment]
I don't see any issues after restarting any of my test machines running macOS 11.3.1. The kernel extension loads without any issues.

**Comment by Chasapple4** [comment]
> I don't see any issues after restarting any of my test machines running macOS 11.3.1. The kernel extension loads without any issues.

Strange for me it will not load and sometimes after 12 to 20+ hours of having the machine on macOS might load the system extension.  Is there a terminal command for forcing macOS to load it?

Tried just now (few days later) and can't load a Win 10 VM (it BSOD or greens screens) because macFUSE system extension is not loaded (it is already approved)

**Comment by bfleischer** [comment]
> Is there a terminal command for forcing macOS to load it?

No, there is not. You can simply ask macOS to load the kernel extension. This is what macFUSE does. The terminal command you are looking for is `kmutil`. 

> Tried just now (few days later) and can't load a Win 10 VM (it BSOD or greens screens) because macFUSE system extension is not loaded (it is already approved)

macFUSE is not responsible for your Windows 10 VM crashing. macFUSE is a macOS kernel extension, not a Windows kernel extension. macFUSE is not loaded inside your VM.

**Comment by Chasapple4** [comment]
@bfleischer what would be the terminal command to tell macOS to load it?

macFuse not loading is the reason for the Win 10 VM crashing (for me not having macFuse loading prevents me from booting any Win 10 VM) VirtualBox uses macFuse (they have an older OSXFuse in the installer, 3.0.4 if I recall right) 

Oracle Virtaulbox
https://www.virtualbox.org/ticket/19963 & https://forums.virtualbox.org/viewtopic.php?f=8&t=101849

**Comment by bfleischer** [comment]
vboximg-mount is a tool for mounting a VirtualBox virtual disk in macOS when the virtual machine is not running. vboximg-mount is an optional component, that is not required for running virtual machines. 

The links you posted do not indicate that macFUSE might be responsible for your VM crashing.

**Comment by Chasapple4** [comment]
> vboximg-mount is a tool for mounting a VirtualBox virtual disk in macOS when the virtual machine is not running. vboximg-mount is an optional component, that is not required for running virtual machines.
> 
> The links you posted do not indicate that macFUSE might be responsible for your VM crashing.

macFuse not loading is what is causing them to crash, VirtualBox uses macFuse to load the VMs, when macFuse is loaded the VMs don't crash on start up.  The only thing that changed is macFuse not loading.  When it works right now it is hard to restart as then macFuse may not load and macOS might not load it at all (last time it was loaded was about 2 weeks ago).

Already tried a reinstall and the terminal command in recovery mode neither work now.

**Comment by bfleischer** [comment]
macFUSE is a VFS kernel plugin for macOS. It has absolutely nothing to do with virtual machines. macFUSE is used to build file systems for macOS. In VirtualBox's case (vboximg-mount) this allows you to access a VM's virtual disk from within the context of macOS without having to start the VM. That's all. macFUSE is in no way required to start VMs and it is most definitely not to blame for your VMs crashing. If you don't believe me, re-read the VirtualBox documentation or raise the question with the VirtualBox team.

**Comment by Chasapple4** [comment]
> macFUSE is a VFS kernel plugin for macOS. It has absolutely nothing to do with virtual machines. macFUSE is used to build file systems for macOS. In VirtualBox's case (vboximg-mount) this allows you to access a VM's virtual disk from within the context of macOS without having to start the VM. That's all. macFUSE is in no way required to start VMs and it is most definitely not to blame for your VMs crashing. If you don't believe me, re-read the VirtualBox documentation or raise the question with the VirtualBox team.

On macOS Virtuabox uses it to start the VM (mainly Windows 10), I have done the terminal command to list system extensions and macFuse is not being allowed to load.  When macFuse is not loaded I get a BSOD in Windows 10 (Win XP boots but hangs a lot).  This is what I have observed since December of 2020.  Even tho it is approved in macOS, macOS is not letting it load (the supposed fix was pulled from 11.3.x).

That is why I have been asking how to manually force macOS to load it (panic trigger in recover does not work nor does an uninstall and install).

**Comment by bfleischer** [comment]
I'll try one last time. VirtualBox does not use or need macFUSE in order to start virtual machines. I can start a Windows 10 VM without macFUSE even being installed. Your VM crashing is not caused by the macFUSE kernel extension not being loaded. This makes absolutely no sense.

That being said, the macFUSE kernel extension not being loaded and your VM crashing might have a common cause, but macFUSE is definitely not to blame for you VM crashes. If you need further assistance with the VM crashes, please raise the issue with the VirtualBox team. They will confirm that macFUSE is not involved in the boot process of your VM. There is nothing I can do to help you.

> (the supposed fix was pulled from 11.3.x).

No, it was not pulled.

**Comment by Chasapple4** [comment]
> No, it was not pulled.

Others on Twitter mentioned the system extension bug fix was pulled from 11.3 update.

macFuse not being loaded by macOS is why they are crashing (I have not changed any VM settings in years (been using one for 6 years with the same VM settings), when I ran the panic in recovery mode to get macFuse to load back in February that worked until I restarted the computer and macFuse which Virtualbox uses on macOS for both Windows & macOS guests, I remember when they started to use it as you could not install beta version of macOS that used APFS file system, VirtualBox uses an older OSXFuse for this. 

Also I asked about how to use that terminal command to load system extensions

**Comment by slawek-es** [comment]
I have just checked and macFUSE works for me without issue on the recently released macOS version 11.4. Looks like the fix might have landed in that version after all.
